lib/travis/build/script/elm.rb

```diff
@@ -0,0 +1,183 @@
+module Travis
+  module Build
+    class Script
+      class Elm < NodeJs
+        # Default NodeJS version to install
+        DEFAULT_NODE_VERSION = '8.9.1'
+
+        DEFAULTS = {
+          elm: '0.19.0',
+          elm_test: '0.19.0',
+          elm_format: '0.19.0'
+        }
+
+        ELM_TEST_REQUIRED_NODE_VERSION = '6.0.0'
+
+        def export
+          super
+          sh.export 'TRAVIS_ELM_VERSION', elm_version, echo: false
+          sh.export 'TRAVIS_ELM_TEST_VERSION', elm_test_version, echo: false
+          sh.export 'TRAVIS_ELM_FORMAT_VERSION', elm_format_version, echo: false
+        end
+
+        def configure
+          super
+
+          config[:node_js] ||= DEFAULT_NODE_VERSION
+        end
+
+        def announce
+          super
+          sh.cmd 'elm --version', echo: true
+          sh.cmd 'elm-test --version', echo: true
+
+          # elm-format doesn't have --version,
+          # but the first line of `elm-format --help` prints the version
+          sh.cmd 'elm-format --help | head -n 1', echo: true
+        end
+
+        def setup
+          super
+
+          sh.echo 'Elm for Travis-CI is not officially supported, ' \
+          'but is community maintained.', ansi: :green
+          sh.echo 'Please file any issues using the following link',
+            ansi: :green
+          sh.echo '  https://github.com/travis-ci/travis-ci/issues' \
+            '/new?labels=community:elm', ansi: :green
+          sh.echo 'and mention \`@avh4\`, \`@lukewestby\`, \`@stoeffel\` and \`@rtfeldman\`' \
+            ' in the issue', ansi: :green
+
+
+          sh.if '! -d sysconfcpus/bin' do
+            install_sysconfcpus
+          end
+
+          sh.fold 'install.elm' do
+            install_elm
+            install_elm_test
+            install_elm_format
+          end
+        end
+
+        def script
+          sh.cmd 'elm-format --validate . && elm-test'
+        end
+
+        def setup_cache
+          if data.cache?(:elm)
+            sh.fold 'cache.elm' do
+              # Cache the ~/.elm directory.
```

Comment:
This comment seems a little off. Seems we can remove this entirely.

Reply:
As in, remove the comment, or some of the code too?

I might be misunderstanding something - is the home directory cached by default?

Elm 0.19 creates a persistent cache that it uses a cross builds, which it puts in `~/.elm`. I want to make sure that gets cached across Travis builds...but maybe I don't need to do anything special to make that happen?